### ISSUE TITLE
XCode 12 build fails with SPM

### ISSUE THREAD

**Issue by Antonito** [marmelroy/PhoneNumberKit#419]
Hello,

I installed this library via Swift Package Manager, and I am facing some issues.
If I build via `swift build`, everything works. But if I generate a `.xcodeproj` and try to build from it, the build fails here: https://github.com/marmelroy/PhoneNumberKit/blob/c7d7b0c91ef0ad7e413022edfcf5d6f75561e2d5/PhoneNumberKit/PhoneNumberKit.swift#L319 (`Type 'Bundle' has no member 'module'`)

Turns out `Bundle+Resources.swift` is missing from the generated dependency.
<img width="311" alt="image" src="https://user-images.githubusercontent.com/11705040/97078234-ebf27c80-15ea-11eb-87cd-057f5ce7067c.png">

If I remove `"Bundle+Resources.swift"` from the excluded files in the `Package.swift`, XCode is able to build my project, but now `swift build` fails.

```
/[...]/.build/checkouts/PhoneNumberKit/PhoneNumberKit/Bundle+Resources.swift:7:16: error: invalid redeclaration of 'module'
    static var module: Bundle = {
               ^
/[...]/.build/x86_64-apple-macosx/debug/PhoneNumberKit.build/DerivedSources/resource_bundle_accessor.swift:4:16: note: 'module' previously declared here
    static var module: Bundle = {
               ^
[1/2] Compiling PhoneNumberKit Bundle+Resources.swift
```

I'm not sure if it's a PhoneNumberKit bug, a SPM one, or if I'm doing something wrong though, any hints?

### Infos
Version: `3.3.1`
XCode: ` 12.0.1 (GM)`
Swift: `5.3.0`

**Comment by iamjono** [comment]
I have the same issue...

**Comment by bukira** [comment]
Fatal error: unable to find bundle named PhoneNumberKit_PhoneNumberKit: file PhoneNumberKit/resource_bundle_accessor.swift, line 27
2021-03-10 19:53:56.012649+0000 xctest[42853:26406992] Fatal error: unable to find bundle named PhoneNumberKit_PhoneNumberKit: file PhoneNumberKit/resource_bundle_accessor.swift, line 27

same

**Comment by paulofierro** [comment]
I can successfully add PhoneNumberKit via SPM and use it in my iOS app. 

However, I do run into this same issue when running unit tests. This is my workaround:

```swift
private lazy var phoneNumberKit: PhoneNumberKit = {
    if isRunningUnitTests() {
        // This is a workaround for a "Bundle not found" issue
        // when attempting to load the metadata json from the
        // framework bundle while running unit tests.
        return PhoneNumberKit(metadataCallback: { nil })
    } else {
        return PhoneNumberKit()
    }
}()
```
`isRunningUnitTests()` returns true if an environment variable in the test plan is set.

**Comment by nicolas255** [comment]
Hello,

I'm also having this crash when trying to preview stuff as long as there is a PhoneNumberKit involved. And this looks exactly as an issue we encountered when we tried to use a single package architecture to manage the resources in our projects.

To put it simply, for some reason, the preview fails to find the bundle where the JSON is. There is a thread about it there: https://developer.apple.com/forums/thread/664295 where [a workaround](https://developer.apple.com/forums/thread/664295#657852022) has been shared.
I'm not sure of the implication of using such a workaround for PhoneNumberKit but it's working fine for us. It would replace what is inside of [`Bundle+Resources.swift`](https://github.com/marmelroy/PhoneNumberKit/blob/master/PhoneNumberKit/Bundle%2BResources.swift).

I really don't have much time this month, but if I can, I'll try to see if it fixes the issue this weekend or the next one. But if somebody can try before that, that would be nice too 🙂.

For now, I'll try to fix this by adding the JSON file in my resources package and load it manually [that way](https://github.com/marmelroy/PhoneNumberKit/issues/247#issuecomment-530663058) (because I think spreading a flag everywhere to know if I'm testing/previewing is kind of not great 😕).
